x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportPutFollowAction.java

```diff
@@ -211,6 +244,29 @@ private void initiateFollowing(
         ));
     }
 
+    static DataStream updateLocalDataStream(Index backingIndexToFollow,
+                                            DataStream localDataStream,
+                                            DataStream remoteDataStream) {
+        if (localDataStream == null) {
+            // The data stream and the backing indices have been created and validated in the remote cluster,
+            // just copying the data stream is in this case safe.
+            return new DataStream(remoteDataStream.getName(), remoteDataStream.getTimeStampField(),
```

Comment:
I think we can't use the generation from the remoteDataStream as it will break the [assertion](https://github.com/elastic/elasticsearch/blob/709264ac4e932fb54c0cad02f06f6b69c29b3ce9/server/src/main/java/org/elasticsearch/cluster/metadata/DataStream.java#L60) in DataStream's constructor. For example, here we are following .ds-1 and the ds generation on the remote is 2 already.

Reply:
That assertion will be removed: https://github.com/elastic/elasticsearch/pull/61525/files#diff-c1a5087d37ad81d4b2339167eadc38bcee234a4f7d646b7b4b193ae665075334L60

This to support migrating write alias with indices to a data stream. It was added as sanity check
initially, because backing indices were created by Elasticsearch exclusively.

Reply:
Ok